### PR TITLE
add screening

### PR DESCRIPTION
**Pull request by ridzikowski**
move screening from travel branch

### CODE REVIEW

File: src/simparams.jl [hunk 1]
```diff
@@ -42,10 +42,12 @@ struct SimParams
   phone_tracing_params::Union{Nothing, PhoneTracingParams}
 
   infection_modulation_function::Union{Nothing,InfectionModulation}
+  screening_params
```
Comment (tozanski) [inline src/simparams.jl:45]: inside a `struct` we need all the types specified

File: src/params/screening.jl [hunk 2]
```diff
@@ -0,0 +1,35 @@
+struct ScreeningParams
+    begin_day::Float64
+    precision::Float64
+    frequency::Float64
+  end
+
+
+function screen_children!(rng::AbstractRNG, state::SimState, params::SimParams)
+    for screening_day in params.screening_params.begin_day:params.screening_params.frequency:365
```
Comment (tozanski) [inline src/params/screening.jl:9]: design: but maybe it should not be a function of ::SimParams
- this does not check if `params.screening === nothing`.
- 365 is a hardcoded value

File: src/params/screening.jl [hunk 3]
```diff
@@ -0,0 +1,35 @@
+struct ScreeningParams
+    begin_day::Float64
+    precision::Float64
+    frequency::Float64
```
Comment (tozanski) [inline src/params/screening.jl:4]: is it a frequency or rather a period?

File: src/params/screening.jl [hunk 4]
```diff
@@ -0,0 +1,35 @@
+struct ScreeningParams
+    begin_day::Float64
```
Comment (tozanski) [inline src/params/screening.jl:2]: naming: since it is a `Float64` it is better to name it something like `start_time`, days would be integers

**Review by tozanski — CHANGES_REQUESTED** [review]
I do not see the screening function being invoked anywhere, perhaps it does not belong to this repository if it is used only in Launcher/Callback

File: src/simparams.jl (hunk 1 above)
Comment (ridzikowski) [reply src/simparams.jl:45]: ok, i added

File: src/params/screening.jl (hunk 3 above)
Comment (ridzikowski) [reply src/params/screening.jl:4]: ok, i change

File: src/params/screening.jl (hunk 4 above)
Comment (ridzikowski) [reply src/params/screening.jl:2]: ok, i change

File: src/params/screening.jl (hunk 2 above)
Comment (ridzikowski) [reply src/params/screening.jl]: I removed hardcoded value and I added check, but fuction is still a part ::SimParams

File: src/event.jl [hunk 5]
```diff
@@ -31,6 +31,8 @@
   RecoveryEvent
   DeathEvent
 
+  ScreenigEvent
```
Comment (eMaerthin) [inline src/event.jl:34]: typo
```suggestion
  ScreeningEvent
```
```suggestion
  ScreenigEvent
```

File: src/params/screening.jl [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+      if rand(state.rng) >= params.screening_params.precision
+        continue
+      end
+      push!(
+        state.queue, 
+        Event(
+          Val(DetectionEvent),
+          time(event),
+          id,
+          OutsideQuarantineDetction),
```
Comment (eMaerthin) [inline src/params/screening.jl:31]: funny but here also typo (for quite a long anyway...)
```suggestion
          OutsideQuarantineDetection),
```

File: src/params/screening.jl [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+          immediate=true)
+    end
+  end
+end
+
+function add_screening!(state::AbstractSimState, params::AbstractSimParams,time_limit::TimePoint=typemax(TimePoint))
```
Comment (eMaerthin) [inline src/params/screening.jl:37]: missing whitespace between args:
 ```suggestion
function add_screening!(state::AbstractSimState, params::AbstractSimParams, time_limit::TimePoint=typemax(TimePoint))
```

File: src/params/screening.jl [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+  for screening_time in 0.0:params.screening_params.period:time_limit
```
Comment (eMaerthin) [inline src/params/screening.jl:38]: `start_time` should be used here
```suggestion
  for screening_time in params.screening_params.start_time:params.screening_params.period:time_limit
```

File: src/simparams.jl [hunk 9]
```diff
@@ -76,6 +78,7 @@ function load_params(rng=MersenneTwister(0);
         population::DataFrame,
         infection_modulation_name::Union{Nothing,AbstractString}=nothing,
         infection_modulation_params::NamedTuple=NamedTuple{}(),
+        screening_params::Union{Nothing,ScreeningParams}=nothing,
```
Comment (eMaerthin) [inline src/simparams.jl:81]: I think we could "hide" `screening_params` in `kwargs`, any reason why we are here initializing it explicitly to `nothing`?

**Review by eMaerthin — CHANGES_REQUESTED** [review]

File: src/event.jl (hunk 5 above)
Comment (ridzikowski) [reply src/event.jl:34]: ok

File: src/params/screening.jl (hunk 6 above)
Comment (ridzikowski) [reply src/params/screening.jl:31]: ok

File: src/params/screening.jl (hunk 7 above)
Comment (ridzikowski) [reply src/params/screening.jl:37]: ok

File: src/params/screening.jl (hunk 8 above)
Comment (ridzikowski) [reply src/params/screening.jl:38]: good suggestion, i missed it

File: src/simparams.jl (hunk 9 above)
Comment (ridzikowski) [reply src/simparams.jl]: ok, i corrected it. probably it was remains after my previous version of the implementation

**Review by eMaerthin — APPROVED** [review]

**Review by tozanski — COMMENTED** [review]
We can merge this as a temporary solution before moving to Callback in `MotionSimLauncher`. In order to do this we need to increase version number in Project.toml

**Pull request merged**